fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/jbatch/bulkdata/load/ChunkWriter.java

```diff
@@ -183,41 +183,44 @@ public void writeItems(List<java.lang.Object> arg0) throws Exception {
-                    if (Constants.IMPORT_IS_COLLECT_OPERATIONOUTCOMES && operationOutcome != null) {
-                        FHIRGenerator.generator(Format.JSON).generate(operationOutcome, chunkData.getBufferStreamForImport());
-                        chunkData.getBufferStreamForImport().write(Constants.NDJSON_LINESEPERATOR);
-                    }
-                } catch (FHIROperationException e) {
-                    logger.warning("Failed to import '" + fhirResource.getId() + "' due to error: " + e.getMessage());
-                    failedNum++;
-                    if (Constants.IMPORT_IS_COLLECT_OPERATIONOUTCOMES) {
-                        OperationOutcome operationOutCome = FHIRUtil.buildOperationOutcome(e, false);
-                        FHIRGenerator.generator(Format.JSON).generate(operationOutCome, chunkData.getBufferStreamForImportError());
-                        chunkData.getBufferStreamForImportError().write(Constants.NDJSON_LINESEPERATOR);
+        txn.begin();
+        try {
+            for (Object objResJsonList : arg0) {
+                @SuppressWarnings("unchecked")
+                List<Resource> fhirResourceList = (List<Resource>) objResJsonList;
+    
+                for (Resource fhirResource : fhirResourceList) {
+                    try {
+                        String id = fhirResource.getId();
+                        processedNum++;
+                        // Skip the resources which failed the validation
+                        if (failValidationIds.contains(id)) {
+                            continue;
+                        }
+                        OperationOutcome operationOutcome =
+                                fhirPersistence.update(persistenceContext, id, fhirResource).getOutcome();
+                        succeededNum++;
+                        if (Constants.IMPORT_IS_COLLECT_OPERATIONOUTCOMES && operationOutcome != null) {
+                            FHIRGenerator.generator(Format.JSON).generate(operationOutcome, chunkData.getBufferStreamForImport());
+                            chunkData.getBufferStreamForImport().write(Constants.NDJSON_LINESEPERATOR);
+                        }
+                    } catch (FHIROperationException e) {
+                        logger.warning("Failed to import '" + fhirResource.getId() + "' due to error: " + e.getMessage());
+                        failedNum++;
+                        if (Constants.IMPORT_IS_COLLECT_OPERATIONOUTCOMES) {
+                            OperationOutcome operationOutCome = FHIRUtil.buildOperationOutcome(e, false);
+                            FHIRGenerator.generator(Format.JSON).generate(operationOutCome, chunkData.getBufferStreamForImportError());
+                            chunkData.getBufferStreamForImportError().write(Constants.NDJSON_LINESEPERATOR);
+                        }
                     }
                 }
             }
+        } finally {
+            // Release the DB connection.
```

Comment:
not sure if this can work well, need to be verified by the minio enabled build pipeline.